src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
 
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[])new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == values.length) {
+            extend();
+        }
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index >= values.length) {
+            throw new java.lang.ArrayIndexOutOfBoundsException();
+        }
```

Comment:
let's make a separate method for this check

Reply:
what if i want to add element to the last index which will be equal values.length